src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java

```diff
@@ -104,10 +130,62 @@ public void createFeedSourceWithNoUrlTest() {
         assertEquals(0, jobCountForFeed(feedSourceWithNoUrl.id));
     }
 
+
+    /**
+     * Create some labels, add them to the feed source make them admin only, and check that they don't appear if not an admin
+     */
+    @Test
+    public void createFeedSourceWithLabels() throws IOException {
+        // Create labels
+        HttpResponse createFirstLabelResponse = TestUtils.makeRequest("/api/manager/secure/label",
+                JsonUtil.toJson(publicLabel),
+                HttpUtils.REQUEST_METHOD.POST
+        );
+        assertEquals(OK_200, createFirstLabelResponse.getStatusLine().getStatusCode());
+        HttpResponse createSecondLabelResponse = TestUtils.makeRequest("/api/manager/secure/label",
+                JsonUtil.toJson(adminOnlyLabel),
+                HttpUtils.REQUEST_METHOD.POST
+        );
+        assertEquals(OK_200, createSecondLabelResponse.getStatusLine().getStatusCode());
+
+        String firstLabelId = publicLabel.id;
+        String secondLabelId = adminOnlyLabel.id;
+
+        feedSourceWithLabels.labels.add(firstLabelId);
+        feedSourceWithLabels.labels.add(secondLabelId);
+
+        // Create feed source with labels
+        HttpResponse createFeedSourceResponse = TestUtils.makeRequest("/api/manager/secure/feedsource",
+                JsonUtil.toJson(feedSourceWithLabels),
+                HttpUtils.REQUEST_METHOD.POST
+        );
+        assertEquals(OK_200, createFeedSourceResponse.getStatusLine().getStatusCode());
+        // Test that they are assigned properly
+        assertEquals(2, labelCountForFeed(feedSourceWithLabels.id));
+        assertEquals(2, Persistence.projects.getById(feedSourceWithLabels.retrieveProject().id).retrieveProjectLabels().size());
```

Comment:
Can you also make this into a helper method:

```suggestion
        assertEquals(2, labelCountForFeedSourceProject(feedSourceWithLabels));
```

Reply:
Refactored in 9ac824a